solr/core/src/java/org/apache/solr/search/SolrCache.java

```diff
@@ -229,31 +227,36 @@ default boolean isRecursionSupported() {
     return this;
   }
 
+  /**
+   * Returns the segmentMap for the searcher over which this cache is valid (cache is informed of
+   * this via {@link #initialSearcher(SolrIndexSearcher)} or {@link #warm(SolrIndexSearcher,
+   * SolrCache)}).
+   */
+  default SegmentMap getSegmentMap() {
```

Comment:
To my understanding, this is to pave the way for segment aware cache? However, at `SolrCache` level, this seems to be a bit too specific? In particular, most cache implementation does not really care about segment allocation?

Not blocker, but something to consider for upstream 😊